src/services/db/GitFileSystemService.ts

```diff
@@ -337,6 +338,86 @@ export default class GitFileSystemService {
     )
   }
 
+  /**
+   * Wrapper over `git diff --name-only` that also creates a local tracking branch for `base` and/or `head` if they do not exist.
+   * Note that `base` and `head` should be branch names, not commit hashes or other usually valid arguments for `git diff`.
+   */
+  getFilesChanged(
+    repoName: string,
+    base = "master",
+    head = "staging"
+  ): ResultAsync<string[], GitFileSystemError> {
+    if (base === head) {
+      return okAsync([])
+    }
+    return this.createLocalTrackingBranchIfNotExists(repoName, base)
+      .andThen(() => this.createLocalTrackingBranchIfNotExists(repoName, head))
+      .andThen(() => {
+        const efsVolPath = this.getEfsVolPathFromBranch(head)
+        return ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+            .diff([`${base}..${head}`, "--name-only"]),
+          (error) => {
+            logger.error(
+              `Error when getting diff files between "${base}" and "${head}": ${error}, when trying to access ${efsVolPath}/${repoName}`
+            )
+
+            if (error instanceof GitError) {
+              return new GitFileSystemError(
+                "Unable to retrieve git diff info from disk"
+              )
+            }
+
+            return new GitFileSystemError("An unknown error occurred")
+          }
+        ).map((files) =>
+          files
+            .trim()
+            .split("\n")
+            .filter((file) => file !== "")
+        )
+      })
+  }
+
+  /**
+   * Get latest commit for a file path on a branch (including deleted files)
+   */
+  getLatestCommitOfPath(
+    repoName: string,
+    path: string,
+    branch = "staging"
+  ): ResultAsync<DefaultLogFields & ListLogLine, GitFileSystemError> {
+    const efsVolPath = this.getEfsVolPathFromBranch(branch)
+    return ResultAsync.fromPromise(
+      this.git
+        .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
+        .log(["-1", branch, "--", path]), // adding -- allows us to get logs even for deleted files
```

Comment:
what's -1?

Reply:
it returns only the latest commit
`https://git-scm.com/docs/git-log#Documentation/git-log.txt--ltnumbergt`

technically not necessary since I'm only returning `logs.latest` below which is functionally the same, but just thought it's clearer

Reply:
can, just add as comments for other engs